core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -32,6 +32,7 @@
  */
 public abstract class RetryingClient<I extends Request, O extends Response>
         extends SimpleDecoratingClient<I, O> {
+    private static final int DEFAULT_MAX_ATTEMPTS = 5;
```

Comment:
I'm not sure if it's a good idea to use a client option for this because it's used only when RetryingClient is in the decorator stack. How about allowing a user override this value by passing a constructor parameter when instantiating RetryingClient (and its subtypes)?